modules/global/src/com/haulmont/cuba/core/sys/AbstractViewRepository.java

```diff
@@ -120,8 +121,15 @@ protected void checkDuplicates(Element rootElem) {
             String key = getMetaClass(viewElem) + "/" + viewName;
             if (!Boolean.parseBoolean(viewElem.attributeValue("overwrite"))) {
                 String extend = viewElem.attributeValue("extends");
-                if (!StringUtils.equals(extend, viewName) && checked.contains(key)) {
-                    log.warn("Duplicate view definition without 'overwrite' attribute and not extending parent view: " + key);
+                if (extend != null) {
+                    List<String> ancestors = Arrays.asList(extend.split(","));

```

Comment:
Need protection from spaces and duplicate commas, e.g. "view1, view2,, view3".
Use com.google.common.base.Splitter: `Splitter.on(',').omitEmptyStrings().trimResults().splitToList(value)`